### ISSUE TITLE
license for this repo.

### ISSUE THREAD

**Issue by divyanshj16** [Guajevilla/PointNet#1]
Your Xconv implementation is very easy to understand unlike other implementations out there. I want to use this implementation for my work on segmentation, can I know the license for this repo?

Also, can you help me figure out the way to increase the number of points/ upsample for the decoder part of the segmentation network and how can we include lateral features in these?

**Comment by Guajevilla** [comment]
Thank you for reminding me. I have just updated the MIT license for this implementation.
Please enjoy.

About upsampling, I'm also confused. In the paper and the author's tensorflow implementation, they both used the same code as xconv for deconv but with different parameters. I am assuming that by setting the output points' number bigger than the input and then combine it with downsampling features is what the author means. However, I haven't proved this idea yet. Please inform me if the idea is right.

**Comment by divyanshj16** [comment]
Thanks for adding the license.

![image](https://user-images.githubusercontent.com/16683472/69778121-2714e180-11c9-11ea-9ae2-61f27d3deb43.png)
As you can see, your xconv module is able to increase the number of points. Is this how you intended it to behave?

How can should we combine it with features from the earlier layer? Should I just add?

Also, I haven't seen the depthwise seperable conv in your implementation which is mentioned in the paper, here.
![image](https://user-images.githubusercontent.com/16683472/69781086-0b163d80-11d3-11ea-9e64-1636afc96bb1.png)

**Comment by Guajevilla** [comment]
Exactly, but the output channel should be lesser, the parameter can be obtained in tensorflow implementation. It's roughly symmetric to the conv process. 
This feature is not added to former features but concatnated, with a following dense connnect.

I haven't understood the process of this depthwise seperable conv yet, so in this implement I just used FC layer.

**Comment by divyanshj16** [comment]
I was going through your code, but couldn't understand what you want to do [here](https://github.com/Guajevilla/PointNet/blob/d614dd9c566aa55c66f937636b28b9418a298e1d/PointCNN.py#L186).
TODO: ind会越界 I cannot understand what this means.

**Comment by divyanshj16** [comment]
Also, can you explain a bit how 
`FC(3∗K, K∗K)→ELU→BN` is equivalent to your
 `            nn.Conv2d(3, K, kernel_size=1) ->
            nn.ELU(inplace=True) ->
            nn.BatchNorm2d(K)`
kernel_size should be (1,k)

**Comment by LiDaiY** [comment]
Hi, @divyanshj16 
Could you tell me the training speed and memory usage when you train the seg model ?
I imitate the configuration of the official to implement the segmentation model. 
But the training was extremely slow and need a lof of memory. The number of parameters is about the same as pointnet++, while the speed is about four times slower than pointnet++，and about four times memory consumption than it. This makes me distressed, so can you tell me the speed and memory usage when you train ?

**Comment by divyanshj16** [comment]
The training speed will heavily depend on the amount of data you have and the number of points in each block.
In PointCNN, for 8192 points per block and a batch size of 12, the GPU memory required was ~32G.
This is for the segmentation model.
